### PR TITLE
Epmlabsbrn-146 prepare for series 2

### PR DESCRIPTION
**Pull request by ElenaSpb**
Epmlabsbrn 146

add preporation code for 2 series: enums, dtos, functions, api raml description, fix tests, some refactoring

### CODE REVIEW

File: src/main/kotlin/com/epam/brn/service/TaskService.kt [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
+    val exerciseRepository: ExerciseRepository
+) {
 
     private val log = logger()
 
-    fun getAllTasksByExerciseId(exerciseId: Long): List<TaskDtoForSingleWords> {
-        val tasks = taskRepository.findAllTasksByExerciseIdWithJoinedAnswers(exerciseId)
-        return tasks.map { task -> task.toSingleWordsDto() }
+    fun getTasksByExerciseId(exerciseId: Long): List<Any> {
+        val exercise: Exercise = exerciseRepository.findById(exerciseId).get()
```
Comment (odanc) [inline src/main/kotlin/com/epam/brn/service/TaskService.kt]: Unsafe `get()` on `Optional`

File: src/main/kotlin/com/epam/brn/service/TaskService.kt [hunk 2]
```diff
@@ -1,35 +1,41 @@
...
 import com.epam.brn.model.Task
+import com.epam.brn.repo.ExerciseRepository
 import com.epam.brn.repo.TaskRepository
 import org.apache.logging.log4j.kotlin.logger
 import org.springframework.stereotype.Service
 
 @Service
-class TaskService(private val taskRepository: TaskRepository) {
+class TaskService(
+    val taskRepository: TaskRepository,
```
Comment (odanc) [inline src/main/kotlin/com/epam/brn/service/TaskService.kt]: Let's keep things `private`

File: src/main/kotlin/com/epam/brn/service/TaskService.kt (hunk 1 above)
Comment (ElenaSpb) [reply src/main/kotlin/com/epam/brn/service/TaskService.kt]: fixed

File: src/main/kotlin/com/epam/brn/service/TaskService.kt (hunk 2 above)
Comment (ElenaSpb) [reply src/main/kotlin/com/epam/brn/service/TaskService.kt]: ok

**Pull request merged**